src/Storages/StorageIceberg.h

```diff
@@ -0,0 +1,64 @@
+#pragma once
+
+#include "config.h"
+
+#if USE_AWS_S3
+
+#    include <Storages/IStorageDataLake.h>
+
+#    include <unordered_map>
+#    include <base/JSON.h>
+
+namespace Poco
+{
+class Logger;
+}
+
+namespace Aws::S3
+{
+class S3Client;
+}
+
+namespace DB
+{
+
+// Class to parse iceberg metadata and find files needed for query in table
+// Iceberg table directory outlook:
+// table/
+//      data/
+//      metadata/
+// The metadata has three layers: metadata -> manifest list -> manifest files
+class IcebergMetaParser
+{
+public:
+    IcebergMetaParser(const StorageS3::S3Configuration & configuration_, const String & table_path_, ContextPtr context_);
```

Comment:
Here it is better to get rid of s3 too and make it abstract. May be use reference to a base class I added...
https://github.com/ClickHouse/ClickHouse/blob/ab0dedf0c88cbfb732d7c0c71922dce49a227607/src/Storages/StorageConfiguration.h ? (Merge this branch into your PR?)